### ISSUE TITLE
[UI] Cluster Templates - Creating azure template with cloud provider config without providing all the credentials should be allowed

### ISSUE THREAD

**Issue by soumyalj** [rancher/rancher#21759]
Version: Master-head

Steps:
1. Create a cluster template/revision with azure cloud provider set and do not provide the credentials .
The credentials are made required currently. Users may not want to provide all the credentials during template creation. They will want to provide them during cluster creations. 
Making them required will not allow us to create a template where credentials can be questions.

![Screen Shot 2019-07-24 at 3.20.00 PM.png](https://zube.io/files/rancher/3c3497823596cad7e7a2a83f59451275-screen-shot-2019-07-24-at-3-20-00-pm.png)

**Comment by davidnuzik** [comment]
This will revert a change recently made in this area, but now that I think of it it makes sense.
Today, if we set questions for all of the four required fields and insert a value it's ignored anyway -- the user leveraging the template must fill out all anyway. If we plug in the values but DON'T make the fields questions then we can leverage the template and it will work (Azure cloud features work). This makes sense and I think we should go forward with the change.

**Comment by davidnuzik** [comment]
Currently blocked by https://github.com/rancher/rancher/issues/22038 would like to get that fixed before I attempt to validate.

**Comment by davidnuzik** [comment]
Version: master-head (8/7/19)

None of the azure cloud credentials are required now when creating a CT. This is good and working as expected so the issue is validated.

I created a separate issue here: https://github.com/rancher/rancher/issues/22041

Ideally we should check if an override is set. If an override is set, then the field is not required. If an override is NOT set for one of the four required fields, then you should NOT be allowed to create the CT (because it will be nonfunctional). That issue addresses that concern.
